### PR TITLE
test(binding-coap): add test for IPv6 support

### LINKED ISSUE #363: EINVAL error on global IPv6 addresses
I used the [node-wot-boilerplate](https://github.com/Citrullin/wot-node-boilerplate) project. Instead of using a link-local address as mentioned in #362, I used a global IPv6 address. For example `2a01:c10a:d6ca:c212:9c7c:1bff:fe90:8f54`
When I execute `yarn dev`, I get the following error
```
yarn dev
yarn run v1.22.5
$ ts-node src/index.ts
[core/helpers] Helpers found scheme 'coap'
[core/servient] Servient creating client for scheme 'coap'
[binding-coap] CoapClientFactory creating client for 'coap'
[core/helpers] WoTImpl fetching TD from 'coap:[2a01:c10a:d6ca:c212:9c7c:1bff:fe90:8f54]/.well-known/wot-thing-description' with [CoapClient]
[binding-coap] CoapClient got Form 'contentType' application/td+json
[binding-coap] CoapClient sending GET to coap:[2a01:c10a:d6ca:c212:9c7c:1bff:fe90:8f54]/.well-known/wot-thing-description
Fetch error: Error: send EINVAL 2a01:c10a:d6ca:c212:9c7c:1bff:fe90:8f54:5683
    at doSend (dgram.js:692:16)
    at defaultTriggerAsyncIdScope (internal/async_hooks.js:429:12)
    at afterDns (dgram.js:638:5)
    at processTicksAndRejections (internal/process/task_queues.js:81:21) {
  errno: -22,
  code: 'EINVAL',
  syscall: 'send',
  address: '2a01:c10a:d6ca:c212:9c7c:1bff:fe90:8f54,
  port: 5683
}
```
I guess the error here is that the IP address is not properly escaped. The parameter should be `[2a01:c10a:d6ca:c212:9c7c:1bff:fe90:8f54]:5683` and not `2a01:c10a:d6ca:c212:9c7c:1bff:fe90:8f54:5683`

### PR DESCRIPTION
**Pull request by JKRhb**
Should resolve #363.

### CODE REVIEW

File: packages/binding-coap/test/coap-server-test.ts [hunk 1]
```diff
@@ -202,6 +202,35 @@ class CoapServerTest {
...
+        });
+
+        const test: DataSchemaValue = "testValue";
+        testThing.setPropertyReadHandler("test", (_) => Promise.resolve(test));
+
+        await coapServer.expose(testThing);
+
+        const uri = `coap://[::1]:${coapServer.getPort()}/test/`;
+
+        const coapClient = new CoapClient();
```
Comment (relu91) [inline packages/binding-coap/test/coap-server-test.ts:226]: I recall that the issue was especially caused by the usage of the server as an argument in the coap client. See:
```ts
// create Servient and add CoAP binding
const servient = new Servient();
servient.addClientFactory(new CoapClientFactory(new CoapServer())); // you have to import CoapServer
```
Can you update the test to closely resemble what would happen if we are using the `CoapClientFactory` ?

**Review by relu91 — CHANGES_REQUESTED** [review]

File: packages/binding-coap/test/coap-server-test.ts (hunk 1 above)
Comment (JKRhb) [reply packages/binding-coap/test/coap-server-test.ts:226]: Oh, sure, in an earlier version I actually reused the socket already, I will update the PR shortly :)

Comment (relu91) [reply packages/binding-coap/test/coap-server-test.ts]: yes, it does!

**Review by relu91 — APPROVED** [review]

File: packages/binding-coap/test/coap-server-test.ts (hunk 1 above)
Comment (JKRhb) [reply packages/binding-coap/test/coap-server-test.ts]: Great! :) Thank you for approving, I took the freedom of squashing the commits into one :)

**Review by danielpeintner — APPROVED** [review]
Thanks @JKRhb
LGTM!

**Pull request merged**